### PR TITLE
feat: add cow amm banner

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Adds a banner for CoW AMM on CoW Swap
- On click of X should close and remember this (localStorage) using existing closable banner component
- On click of CTA button shoudl track GA event

https://github.com/user-attachments/assets/4b15a886-801c-494a-838e-7173b2317130

Screenshot 2024-08-09 at 17 26 22

Screenshot 2024-08-09 at 17 26 29

### CODE REVIEW

File: apps/cowswap-frontend/src/common/pure/CoWAMMBanner/index.tsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  transition: opacity 0.2s ease-in-out;
+
+  &:hover {
+    opacity: 1;
+  }
+`
+
+export function CoWAmmBanner() {
+  const handleCTAClick = () => {
+    cowAnalytics.sendEvent({
```
Comment (anxolin) [inline apps/cowswap-frontend/src/common/pure/CoWAMMBanner/index.tsx:198]: technically not pure component if it has analytics

File: apps/cowswap-frontend/src/common/pure/CoWAMMBanner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      action: 'CoW AMM Banner CTA Clicked',
+    })
+
+    window.open(
+      'https://balancer.fi/pools/cow?utm_source=swap.cow.fi&utm_medium=web&utm_content=cow_amm_banner',
+      '_blank'
+    )
+  }
+
+  return ClosableBanner('cow_amm_banner', (close) => (
```
Comment (anxolin) [inline apps/cowswap-frontend/src/common/pure/CoWAMMBanner/index.tsx:209]: also, closable banners are not pure (they depend on state), however, this was already wrong before this PR (i see ClosableBanner is in the pure directory)

**Review by anxolin — APPROVED** [review]
Looks great, link works, and analytics work

File: apps/cowswap-frontend/src/common/pure/CoWAMMBanner/index.tsx (hunk 2 above)
Comment (alfetopito) [reply apps/cowswap-frontend/src/common/pure/CoWAMMBanner/index.tsx:209]: I personally consider internal state to be fine for pure components.
As long as it doesn't use hooks that load state from outside it's ok for me.

**Pull request merged**